app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,423 @@
+}
+
+const addOpenTooltip = (id: string, dispatcher: React.Dispatch<unknown>) => {
+  if (openTooltips.find(([openId]) => openId === id)) {
+    return
+  }
+
+  openTooltips.push([id, dispatcher])
+}
+
+const removeOpenTooltip = (id: string) => {
+  const index = indexOfOpenTooltip(id)
+  if (index !== -1) {
+    openTooltips.splice(index, 1)
+  }
+}
+
+const closeOtherOpenTooltips = (id: string) => {
+  openTooltips.forEach(([openId, openDispatch]) => {
+    if (openId !== id) {
+      openDispatch({
+        id: openId,
+        action: {
+          type: 'hide',
+        },
+      })
+    }
+  })
+}
+
+const shouldShow = (state: TooltipState): boolean => {
+  return state.focus || state.hover || state.requestFocus || state.requestHover
+}
+
+const handleShowRequest = (state: TooltipState): ShowState => {
+  return shouldShow(state) && state.showState === 'hidden'
+    ? 'loading'
+    : state.showState
+}
+
+const handleHideRequest = (state: TooltipState): ShowState => {
+  return !shouldShow(state) ? 'hidden' : state.showState
+}
+
+/**
+ * tooltipReducer
+ * This serves as a state reducer for the Concept (tooltip) Component for use with
+ * the useReducer hook.
+ */
+const tooltipReducer: ShowReducer = function (state, body) {
+  const { id, action } = body
+  const nextState: TooltipState = { ...state }
+
+  // Use action to update the next state
+  switch (action.type) {
+    case 'request-show-from-ref-hover':
+      nextState.requestHover = true && !hasOpenTooltip(id)
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-from-ref-hover':
+      nextState.requestHover = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-from-ref-focus':
+      nextState.requestFocus = true
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-from-ref-focus':
+      nextState.requestFocus = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-hover':
+      nextState.hover = true && !hasOpenTooltip(id)
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-hover':
+      nextState.hover = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'request-show-focus':
+      nextState.focus = true
+      nextState.showState = handleShowRequest(nextState)
+      break
+    case 'request-hide-focus':
+      nextState.focus = false
+      nextState.showState = handleHideRequest(nextState)
+      break
+    case 'loaded':
+      nextState.showState = 'invisible'
+      break
+    case 'show':
+      nextState.showState = 'visible'
+      break
+    case 'hide':
+      nextState.showState = 'hidden'
+      break
+    case 'error':
+      nextState.showState = 'error'
+      break
+    default:
+      throw new Error('unhandled tooltip reducer action')
+  }
+
+  return nextState
+}
+
+export const Concept = ({
```

Comment:
Just to check: most of the code is not specific to the Concept tooltip, but could be applied to other tooltips too, right? Were you planning to extract this to a separate component, for it to be re-used?

Reply:
We had discussed this in slack, and the one case that @iHiD had been thinking of we weren't sure if it just deserved just another type of tooltip. So I haven't made an active effort to ensure generalization

Reply:
Makes sense!